src/foam/nanos/fs/File.js

```diff
@@ -69,9 +68,63 @@ foam.CLASS({
         }
         return url;
       }
+    },
+    {
+      class: 'Blob',
+      name: 'data',
+      javaGetter:`
+        if ( dataIsSet_ ) return data_;
+        if ( this.getDataString() != null && this.getDataString() != "" ){
```

Comment:
```suggestion
        if ( ! SafetyUtil.isEmpty(this.getDataString()) ) {
```
// foam.util.SafetyUtil.